### PR TITLE
Add Chatterino API Privacy Policy and Terms of Service

### PR DESCRIPTION
**Pull request by pajlada**
- Add privacy policy and terms of service
- stop gitignore from ignoring everything in the cmd/api directory

Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

### CODE REVIEW

**Comment by pajlada** [comment]
@Felanbird Could you do an English check on the text?

File: cmd/api/legal.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	const text = `<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Chatterino API - Terms of Service</title>
+</head>
+<body>
+  <h1>Chatterino API - Terms of Service</h1>
+  <p>
+    Using the Chatterino API from the Chatterino Client is perfectly fine.
```
Comment (Felanbird) [inline cmd/api/legal.go:19]:
```suggestion
    Usage of the Chatterino API from the Chatterino Client is perfectly fine.
```
The original is perfectly fine, I just feel like the word `Usage` is very often used in this sort of thing

**Comment by pajlada** [comment]
pretty links can be found here:
https://pajbot.com/static/chatterino-api-tos.html
https://pajbot.com/static/chatterino-api-pp.html

File: cmd/api/legal.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  </p>
+  <p>
+    The Chatterino API uses the following external APIs to provide users with tooltips:<br>
+    <ul>
+    <li>YouTube Videos and Channels through the YouTube Data API. YouTube's Terms of Service can be found here: <a href="https://www.youtube.com/t/terms">https://www.youtube.com/t/terms</a>. Note that no data is forwarded about you the user to the YouTube Data API.</li>
+    </ul>
+	This list of external APIs is not exhaustive.
+  </p>
+  <p>
+	If you have any questions about this terms of service, you can <a href="https://github.com/Chatterino/api/issues/new">contact us</a>.
```
Comment (zneix) [inline cmd/api/legal.go:30]:
```suggestion
	If you have any questions about these Terms of Service, you can <a href="https://github.com/Chatterino/api/issues/new">contact us</a>.
```
or
```suggestion
	If you have any questions about our Terms of Service, you can <a href="https://github.com/Chatterino/api/issues/new">contact us</a>.
```

**Comment by Felanbird** [comment]
`these` is probably better, it seems to be common to see both `this` and `these` terms of service...

File: cmd/api/legal.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	</ul>
+	Your IP address is <strong>NOT</strong> processed or stored.<br>
+	The version of your Chatterino Client is <strong>NOT</strong> processed or stored, but we maintain the right to process it in the future in case we need to provide information in different formats based on the Chatterino Client version.<br>
+	The link you requested information about is processed and stored together with the output data for up to 24 hours as cache to save processing power/bandwidth for us and any external APIs.<br>
+	No data other than the link is shared with external APIs, which is strictly necessary for the Chatterino API to function.
+  </p>
+  <p>
+    The Chatterino API uses the following external APIs to provide users with tooltips:
+    <ul>
+    <li>YouTube Videos and Channels through the YouTube Data API. Google's Privacy policy can be found here: <a href="https://policies.google.com/privacy">https://policies.google.com/privacy</a>. Note that no data is forwarded about you the user to the YouTube Data API.</li>
```
Comment (Felanbird) [inline cmd/api/legal.go:80]:
```suggestion
    <li>YouTube Videos and Channels through the YouTube Data API. Google's Privacy Policy can be found here: <a href="https://policies.google.com/privacy">https://policies.google.com/privacy</a>. Note that no data is forwarded about you the user to the YouTube Data API.</li>
```

File: cmd/api/legal.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	<li>When the Chatterino Client makes a request to the Chatterino API about a YouTube video link (e.g. https://www.youtube.com/watch?v=7fnTxm1u_34), we will make a request about that video (7fnTxm1u_34) to the YouTube Data API and read data such as the video title, the channel the video is uploaded by, the duration of the video, thumbnail link, and more to provide the user with that data in a tooltip format like this: <pre>William Onyeabor - When the Going is Smooth &amp; Good (Official Audio)
+Channel: Luaka Bop
+Duration: 00:12:55
+Published: 11 Aug 2015
+Views: 425,717
+4,604 likes</pre></li>
+	</ul>
+  </p>
+  <p>
+	If you have any questions about this privacy policy, you can <a href="https://github.com/Chatterino/api/issues/new">contact us</a>.
```
Comment (Felanbird) [inline cmd/api/legal.go:96]:
```suggestion
	If you have any questions about this Privacy Policy, you can <a href="https://github.com/Chatterino/api/issues/new">contact us</a>.
```

**Review by Felanbird — CHANGES_REQUESTED** [review]

**Review by Felanbird — APPROVED** [review]

**Review by zneix — APPROVED** [review]

**Pull request merged**